### PR TITLE
Standardise gUM calls, change 'ended' to 'inactive', etc.

### PR DESCRIPTION
**Pull request by samdutton**
**Description**
Minor tweaks to standardise use of gUM and move from using the stream `ended` event to `inactive`.

Also includes:
• A couple of warning messages about feedback.
• Minor tweaks to code layout.

**Purpose**
As per #788.

@KaptenJansson PTAL. (Sorry — I know this is a really fiddly PR, without much gain!)

### CODE REVIEW

**Comment by fippo** [comment]
did you consider doing it with a codemod? :-)

File: src/content/devices/input-output/js/main.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
 /*
- *  Copyright (c) 2015 The WebRTC project authors. All Rights Reserved.
- *
- *  Use of this source code is governed by a BSD-style license
- *  that can be found in the LICENSE file in the root of the source
- *  tree.
- */
+*  Copyright (c) 2015 The WebRTC project authors. All Rights Reserved.
```
Comment (KaptenJansson) [inline src/content/devices/input-output/js/main.js:2]: 2016

File: src/content/devices/input-output/js/main.js [hunk 2]
```diff
@@ -30,11 +30,11 @@ function gotDevices(deviceInfos) {
     option.value = deviceInfo.deviceId;
     if (deviceInfo.kind === 'audioinput') {
       option.text = deviceInfo.label ||
-        'microphone ' + (audioInputSelect.length + 1);
+      'microphone ' + (audioInputSelect.length + 1);
```
Comment (KaptenJansson) [inline src/content/devices/input-output/js/main.js]: actually these should be indented by 4 right? The linter should complain here.

File: src/content/devices/input-output/js/main.js [hunk 3]
```diff
@@ -30,11 +30,11 @@ function gotDevices(deviceInfos) {
...
       audioInputSelect.appendChild(option);
     } else if (deviceInfo.kind === 'audiooutput') {
       option.text = deviceInfo.label || 'speaker ' +
-          (audioOutputSelect.length + 1);
+      (audioOutputSelect.length + 1);
```
Comment (KaptenJansson) [inline src/content/devices/input-output/js/main.js]: actually these should be indented by 4 right? The linter should complain here.

File: src/content/devices/input-output/js/main.js [hunk 4]
```diff
@@ -71,10 +65,10 @@ function attachSinkId(element, sinkId) {
       var errorMessage = error;
       if (error.name === 'SecurityError') {
         errorMessage = 'You need to use HTTPS for selecting audio output ' +
-            'device: ' + error;
+        'device: ' + error;
```
Comment (KaptenJansson) [inline src/content/devices/input-output/js/main.js]: actually these should be indented by 4 right? The linter should complain here.

File: src/content/devices/input-output/js/main.js [hunk 5]
```diff
@@ -71,10 +65,10 @@ function attachSinkId(element, sinkId) {
...
       }
       console.error(errorMessage);
-      // Jump back to first output device in the list as it's the default.
+// Jump back to first output device in the list as it's the default.
```
Comment (KaptenJansson) [inline src/content/devices/input-output/js/main.js]: actually these should be indented by 4 right? The linter should complain here.

File: src/content/devices/input-output/js/main.js [hunk 6]
```diff
@@ -99,19 +100,16 @@ function start() {
...
-  .then(function(stream) {
-    window.stream = stream; // make stream available to console
-    videoElement.srcObject = stream;
-    // Refresh button list in case labels have become available
-    return navigator.mediaDevices.enumerateDevices();
-  })
-  .then(gotDevices)
-  .catch(errorCallback);
+  navigator.mediaDevices.getUserMedia(constraints).
+  then(gotStream).then(gotDevices).catch(handleError);
```
Comment (KaptenJansson) [inline src/content/devices/input-output/js/main.js]: actually these should be indented by 4 right? The linter should complain here.

File: src/content/devices/multi/js/main.js [hunk 7]
```diff
@@ -101,12 +95,13 @@ function start() {
...
     video: true
   };
-  navigator.mediaDevices.getUserMedia(
-    constraints
-  ).then(
-    successCallback,
-    errorCallback
-  );
+  navigator.mediaDevices.getUserMedia(constraints).
+    then(gotStream).catch(handleError);
```
Comment (KaptenJansson) [inline src/content/devices/multi/js/main.js]: actually these should be indented by 4 right? The linter should complain here.

File: src/content/extensions/desktopcapture/app.js [hunk 8]
```diff
@@ -1,16 +1,18 @@
...
 document.getElementById('start').addEventListener('click', function() {
   // send screen-sharer request to content-script
   if (!extensionInstalled) {
     var message = 'Please install the extension:\n' +
-                  '1. Go to chrome://extensions\n' +
-                  '2. Check: "Enable Developer mode"\n' +
-                  '3. Click: "Load the unpacked extension..."\n' +
-                  '4. Choose "extension" folder from the repository\n' +
-                  '5. Reload this page';
+    '1. Go to chrome://extensions\n' +
```
Comment (KaptenJansson) [inline src/content/extensions/desktopcapture/app.js]: actually these should be indented by 4 right? The linter should complain here.

File: src/content/extensions/desktopcapture/app.js [hunk 9]
```diff
@@ -49,15 +60,7 @@ function startScreenStreamFrom(streamId) {
...
-    videoElement.src = URL.createObjectURL(screenStream);
-    videoElement.play();
-  },
-  // errorCallback
-  function(err) {
-    console.log('getUserMedia failed!: ' + err);
-  });
+  };
+  navigator.mediaDevices.getUserMedia(constraints).
+  then(handleSuccess).catch(handleError);
```
Comment (KaptenJansson) [inline src/content/extensions/desktopcapture/app.js]: actually these should be indented by 4 right? The linter should complain here.

File: src/content/getusermedia/audio/js/main.js [hunk 10]
```diff
@@ -10,29 +10,26 @@
...
 }
 
-navigator.mediaDevices.getUserMedia(
-  constraints
-).then(
-  successCallback,
-  errorCallback
-);
+navigator.mediaDevices.getUserMedia(constraints).
+  then(handleSuccess).catch(handleError);
```
Comment (KaptenJansson) [inline src/content/getusermedia/audio/js/main.js]: actually these should be indented by 4 right? The linter should complain here.

File: src/content/getusermedia/canvas/js/main.js [hunk 11]
```diff
@@ -27,18 +27,14 @@ var constraints = {
...
 }
 
-navigator.mediaDevices.getUserMedia(
-  constraints
-).then(
-  successCallback,
-  errorCallback
-);
+navigator.mediaDevices.getUserMedia(constraints).
+  then(handleSuccess).catch(handleError);
```
Comment (KaptenJansson) [inline src/content/getusermedia/canvas/js/main.js]: actually these should be indented by 4 right? The linter should complain here.

File: src/content/getusermedia/filter/js/main.js [hunk 12]
```diff
@@ -32,18 +31,14 @@ var constraints = {
...
 }
 
-navigator.mediaDevices.getUserMedia(
-  constraints
-).then(
-  successCallback,
-  errorCallback
-);
+navigator.mediaDevices.getUserMedia(constraints).
+  then(handleSuccess).catch(handleError);
```
Comment (KaptenJansson) [inline src/content/getusermedia/filter/js/main.js]: actually these should be indented by 4 right? The linter should complain here.

File: src/content/getusermedia/gum/js/main.js [hunk 13]
```diff
@@ -37,11 +38,14 @@ navigator.mediaDevices.getUserMedia(constraints)
...
 
 function errorMsg(msg, error) {
   errorElement.innerHTML += '<p>' + msg + '</p>';
   if (typeof error !== 'undefined') {
     console.error(error);
   }
 }
+
+navigator.mediaDevices.getUserMedia(constraints).
+  then(handleSuccess).catch(handleError);
```
Comment (KaptenJansson) [inline src/content/getusermedia/gum/js/main.js]: actually these should be indented by 4 right? The linter should complain here.

File: src/content/getusermedia/record/js/main.js [hunk 14]
```diff
@@ -70,22 +60,12 @@ function successCallback(stream) {
...
-//   if (window.URL) {
-//     gumVideo.src = window.URL.createObjectURL(stream);
-//   } else {
-//     gumVideo.src = stream;
-//   }
-// }).catch(function(error) {
-//   console.log('navigator.getUserMedia error: ', error);
-// });
+navigator.mediaDevices.getUserMedia(constraints).
+  then(handleSuccess).catch(handleError);
```
Comment (KaptenJansson) [inline src/content/getusermedia/record/js/main.js]: actually these should be indented by 4 right? The linter should complain here.

File: src/content/getusermedia/resolution/js/main.js [hunk 15]
```diff
@@ -75,11 +75,7 @@ function getMedia(constraints) {
...
   }
   setTimeout(function() {
-    navigator.mediaDevices.getUserMedia(
-      constraints
-    ).then(
-      successCallback,
-      errorCallback
-    );
+    navigator.mediaDevices.getUserMedia(constraints).
+      then(handleSuccess).catch(handleError);
```
Comment (KaptenJansson) [inline src/content/getusermedia/resolution/js/main.js]: actually these should be indented by 4 right? The linter should complain here.

File: src/content/getusermedia/volume/js/main.js [hunk 16]
```diff
@@ -43,22 +43,18 @@ function successCallback(stream) {
     }
     setInterval(function() {
       instantMeter.value = instantValueDisplay.innerText =
-            soundMeter.instant.toFixed(2);
+        soundMeter.instant.toFixed(2);
```
Comment (KaptenJansson) [inline src/content/getusermedia/volume/js/main.js]: actually these should be indented by 4 right? The linter should complain here.

File: src/content/getusermedia/volume/js/main.js [hunk 17]
```diff
@@ -43,22 +43,18 @@ function successCallback(stream) {
...
       slowMeter.value = slowValueDisplay.innerText =
-            soundMeter.slow.toFixed(2);
+        soundMeter.slow.toFixed(2);
```
Comment (KaptenJansson) [inline src/content/getusermedia/volume/js/main.js]: actually these should be indented by 4 right? The linter should complain here.

File: src/content/getusermedia/volume/js/main.js [hunk 18]
```diff
@@ -43,22 +43,18 @@ function successCallback(stream) {
...
 }
 
-navigator.mediaDevices.getUserMedia(
-  constraints
-).then(
-  successCallback,
-  errorCallback
-);
+navigator.mediaDevices.getUserMedia(constraints).
+  then(handleSuccess).catch(handleError);
```
Comment (KaptenJansson) [inline src/content/getusermedia/volume/js/main.js]: actually these should be indented by 4 right? The linter should complain here.

File: src/content/peerconnection/webaudio-input/js/main.js [hunk 19]
```diff
@@ -41,8 +41,7 @@ function start() {
     video: false
   };
   navigator.mediaDevices.getUserMedia(constraints)
-  .then(gotStream)
-  .catch(gotStreamFailed);
+    .then(handleSuccess).catch(handleFailure);
```
Comment (KaptenJansson) [inline src/content/peerconnection/webaudio-input/js/main.js]: actually these should be indented by 4 right? The linter should complain here. Also decide if you want the `.` on the new line or at the end of main method line. Personally I like the `.` on the new line.

File: src/content/peerconnection/webaudio-input/js/main.js [hunk 20]
```diff
@@ -77,30 +74,29 @@ function gotStream(stream) {
...
     pc2.onicecandidate = iceCallback2;
     pc2.onaddstream = gotRemoteStream;
-
     pc1.addStream(filteredStream);
     pc1.createOffer()
-    .then(gotDescription1)
-    .catch(function(error) {
-      console.log('createOffer failed: ' + error);
-    });
+      .then(gotDescription1)
```
Comment (KaptenJansson) [inline src/content/peerconnection/webaudio-input/js/main.js]: actually these should be indented by 4 right? The linter should complain here.

File: src/content/devices/input-output/js/main.js (hunk 1 above)
Comment (samdutton) [inline src/content/devices/input-output/js/main.js:2]: I asked around, and I think we can leave these. Someone else more knowledgeable might disagree though :).

File: src/content/devices/input-output/js/main.js (hunk 2 above)
Comment (samdutton) [inline src/content/devices/input-output/js/main.js]: From Sublime Text, at least, eslint seems to be a bit random about indenting :/.

Fixed.

File: src/content/devices/input-output/js/main.js (hunk 3 above)
Comment (samdutton) [inline src/content/devices/input-output/js/main.js]: Fixed.

File: src/content/devices/input-output/js/main.js (hunk 4 above)
Comment (samdutton) [inline src/content/devices/input-output/js/main.js]: Frustrating... Somehow I wound up reindenting these (and they got messed up in the process).

File: src/content/peerconnection/webaudio-input/js/main.js (hunk 19 above)
Comment (samdutton) [inline src/content/peerconnection/webaudio-input/js/main.js]: Also moved the dot back to the previous line for all these (so the line starts with `then` not `.then`, etc.) Not sure what's best — but it's probably good if we stick to one style.

File: src/content/extensions/desktopcapture/app.js (hunk 8 above)
Comment (samdutton) [inline src/content/extensions/desktopcapture/app.js]: Thanks — fixed in next commit.

File: src/content/getusermedia/volume/js/main.js (hunk 16 above)
Comment (samdutton) [inline src/content/getusermedia/volume/js/main.js]: Done in next commit.

**Comment by samdutton** [comment]
@KaptenJansson  Sorry — missed your other spacing notes. Now fixed, PTAL.

**Comment by KaptenJansson** [comment]
LGTM after rebase.

**Comment by KaptenJansson** [comment]
I will also disable the test that keeps failing.

**Comment by samdutton** [comment]
> LGTM after rebase.

I've merged from upstream gh-pages.

**Pull request merged**